msal/authority.py

```diff
@@ -91,8 +99,9 @@ def __init__(self, authority_url, http_client, validate_authority=True):
             tenant_discovery_endpoint = payload['tenant_discovery_endpoint']
         else:
             tenant_discovery_endpoint = (
-                'https://{}{}{}/.well-known/openid-configuration'.format(
+                'https://{}:{}{}{}/.well-known/openid-configuration'.format(
                     self.instance,
+                    authorityPort,
                     authority.path,  # In B2C scenario, it is "/tenant/policy"
                     "" if tenant == "adfs" else "/v2.0" # the AAD v2 endpoint
```

Comment:
?? [@msal python team] "tenant" having value "adfs" looks like wrong pattern. In ADFS-based azure system there is still a tenantId value that customer would have / know / use. The different is in constructing URLs, the "authority" for adfs endpoints is /adfs whereas for AAD systems it is /tenantId. So customer will instead pass authority value with /adfs segement on it, rather than "hacking" the tenatnId input to provide "adfs" to "cheese" these URI builder methods. And instead, you may ignore tenantId input parameter for these URL building flows, but it may still be relevant in other areas (e.g. caching tokens with tenantId in some cache key perhaps? etc.)

Reply:
The snippet here is not really about "adfs" vs "tenant_id". It is constructing the tenant discovery uri.

Perhaps you can provide a couple examples on an actual scenario (without any chese :-) ) to show where the problem is?

Reply:
Yup so on closer look this looks fine, I got ahead of myself worried about the patterns in CLI which use msal. They have separate parameters for “authority” and “tenantId” and have logic about concatenation them together which causes issues in various parametrized scripts and such.